lumen/ai/tools.py

```diff
@@ -104,45 +104,100 @@ class TableLookup(VectorLookupTool):
     provides = param.List(default=["closest_tables"], readonly=True, doc="""
         List of context values this Tool provides to current working memory.""")
 
+    include_metadata = param.Boolean(default=True, doc="""
+        Whether to include table descriptions in the embeddings and responses.""")
+
+    include_columns = param.Boolean(default=False, doc="""
+        Whether to include column descriptions in the embeddings.""")
+
+    max_concurrent = param.Integer(default=5, doc="""
+        Maximum number of concurrent metadata fetch operations.""")
+
     def __init__(self, **params):
         super().__init__(**params)
+        self._table_metadata = {}
+        self._metadata_tasks = set()  # Track ongoing metadata fetch tasks
+        self._semaphore = asyncio.Semaphore(self.max_concurrent)
         self._memory.on_change('sources', self._update_vector_store)
+
         if sources := self._memory.get("sources"):
             self._update_vector_store(None, None, sources)
         elif source := self._memory.get("source"):
             self._update_vector_store(None, None, [source])
 
+    async def _fetch_and_store_metadata(self, source, table):
+        """Fetch metadata for a table and store it in the vector store."""
+        async with self._semaphore:
+            metadata = await asyncio.to_thread(source.get_metadata, table)
+        multi_source = len(self._memory.get("sources", [])) > 1
+        if multi_source:
+            enriched_text = table_name = f"{source.name}{SOURCE_TABLE_SEPARATOR}{table}"
+        else:
+            enriched_text = table_name = table
+        self._table_metadata[table_name] = metadata
+
+        vector_metadata = {"table": table, "source": source.name}
```

Comment:
What is `table` here? Should it be table_name Maybe add a comment if this is intentional.

Also, semi confusing it's called table_name some places, and source_table elsewhere.